core/string/src/type.rs

```diff
@@ -0,0 +1,90 @@
+//! Module containing string types public and crate-specific.
+use crate::vtable::SequenceString;
+use crate::{JsStr, JsStringKind};
+use std::alloc::Layout;
+
+pub(crate) mod sealed {
```

Comment:
nit: remove `pub(crate)`

I know it's being imported below, but I think it's better to fully qualify the path, i.e. `sealed::Sealed` and `sealed::InternalStringType`. To me, it's a bit higher signal.